griptape/artifacts/base_artifact.py

```diff
@@ -1,15 +1,20 @@
 from __future__ import annotations
 from griptape.mixins import SerializableMixin
-from typing import Any
+from typing import Any, TYPE_CHECKING, Optional
 import json
 import uuid
 from abc import ABC, abstractmethod
 from attrs import define, field, Factory
 
+if TYPE_CHECKING:
+    from griptape.common import Reference
 
-@define()
+
+@define
 class BaseArtifact(SerializableMixin, ABC):
     id: str = field(default=Factory(lambda: uuid.uuid4().hex), kw_only=True, metadata={"serializable": True})
+    reference: Optional[Reference] = field(default=None, kw_only=True, metadata={"serializable": True})
```

Comment:
Just to make sure I understand, `reference` will only be populated for excerpts loaded directly from the source (or vector db).  Artifacts that are derived from source(s) will not have `reference` populated, but will instead contain footnotes directly in the artifact content. Right?